### ISSUE TITLE
No learning_starts threshold: the first gradient steps resample the same 33 opening transitions

### ISSUE THREAD

**Issue by eriiiko** [Apelsinmack/Civulator#80]
Found while working out the replay-ratio arithmetic for #22 (`docs/training_throughput_design.md` §4B). Independent of that design — it affects every run we have already done.

## The gate

`civulator/training/trainer.py:325`:

```python
if len(current_agent.memory) > batch_size:
    current_agent.optimize(batch_size)
```

`batch_size` is 32 (`config.toml [training]`). So optimization begins as soon as the buffer holds **33 transitions**, and `compute_loss` immediately samples 32 of them.

## Why that is a problem

For the first few hundred updates the agent is repeatedly sampling almost the entire buffer — the same handful of moments from the opening of the first game — and taking a gradient step on them every single action. The batch is not an approximately-independent sample of experience; it is nearly the same 33 correlated transitions over and over, drawn from a near-random policy on turn 1 of one map.

Two known failure modes converge exactly here:

- **Primacy bias** ([Nikishin et al., ICML 2022](https://arxiv.org/abs/2205.07802)): deep RL agents overfit early interactions and then discount later evidence for the rest of training. Hammering the first 33 transitions is the maximal-strength version of the setup that paper describes.
- **Correlated-batch instability** — the reason experience replay exists at all is to decorrelate consecutive transitions. At `len(memory) ≈ batch_size` replay provides essentially none of that decorrelation.

Classic DQN avoids this with a **learning-starts** threshold: collect a large number of transitions under the exploration policy *before* the first gradient step (50k in the original Atari setup). We have no such threshold — `> batch_size` is a guard against sampling more transitions than exist, not a warmup.

## Proposed fix

- Add `[training] learning_starts` to `config.toml`; gate becomes `len(memory) >= learning_starts`.
- Default: a real fraction of the 10,000-capacity buffer. 1000 is a defensible starting point (10% of capacity, ~2–3 episodes of experience before the first update) — the value itself is worth a quick ladder rather than a guess.
- `BuildAgent.optimize` has the same shape and should take the same treatment (`trainer.py:246`).

## Scope and comparability

**This is not bit-identical and it is not neutral.** It changes what every future run learns from in its opening phase, so it is a training-dynamics change like any other: it needs a before/after evaluation against the frozen #39 baseline, a CHANGELOG note, and a manifest that records `learning_starts`.

Its natural home is the same patch series as the replay-ratio ladder in #22 §4B/§9, since both are one-line changes at the same call site and both need the same fixed-wall-clock comparison. Keeping them as **separate rungs** of that ladder matters though — one variable at a time.

## Not claimed

No measurement here shows that this has damaged existing results. The argument is mechanistic (the code path) plus published evidence about the failure mode, not an observed regression in our stats. Establishing whether it actually costs us anything is what the before/after evaluation is for.
